### ISSUE TITLE
Don't highlight mention of unknown user

### ISSUE THREAD

**Issue by schiessle** [nextcloud/server#10021]
Follow up from here: https://github.com/nextcloud/spreed/issues/1006

This is what happens if I mention a user who doesn't exists:

![talk - mention](https://user-images.githubusercontent.com/1589737/41961146-eff23e98-79f1-11e8-9051-c73a3c23b731.png)

I think instead we shouldn't format the mention and show it as a normal string. Imagine I want to tell someone `I'm @schiessle on Twitter`, this shouldn't be changed to `I'm @UnknownUser at Twitter`

Some additional thoughts from @jancborchardt : 

Agree with @schiessle here. There’s some other cases additionally:
- You simply have a typo in the username: `@schiesle` or `@blizzzz` would be completely made unreadable to `@Unknown user`, which is worse than showing them with a typo
- The case of LDAP you bring up we can easily catch as the LDAP ID is a format we can parse. For those non-existent LDAP users we can indeed show `@Unknown user`
- In the case where you do _not_ use LDAP but Nextcloud user management and a user is deleted, then showing `@Unknown user` will make the context invisible too. With a non-marked-up `@ivan` you at least know who it was addressed too, even though they aren’t in the system anymore.

**Comment by schiessle** [comment]
cc @mario @blizzz let's continue here

**Comment by blizzz** [comment]
> You simply have a typo in the username: @schiesle or @blizzzz would be completely made unreadable to @Unknown user, which is worse than showing them with a typo

You don't know usernames. Using auto-completion takes away this burden. 

> The case of LDAP you bring up we can easily catch as the LDAP ID is a format we can parse. For those non-existent LDAP users we can indeed show @Unknown user

⛔ No workarounds

> In the case where you do not use LDAP but Nextcloud user management and a user is deleted, then showing @Unknown user will make the context invisible too. With a non-marked-up @ivan you at least know who it was addressed too, even though they aren’t in the system anymore.

True. Until a new Ivan shows up ;) 

But perhaps 
* Ivan has been Michelle a couple years ago before fixing gender, just the user id stays with ``@michelle``
* or just the last name was used as user ID and who once was Santiago (``@santiago``) is perhaps Ruiz for two decades due to marriage an is only known as such by most colleagues. Who then knows who ``@santiago`` was? 

**User IDs are fixed and cannot be changed.**

Anyway, I already said that I acknowledge the problem and gave one suggestion how to solve the issue in https://github.com/nextcloud/spreed/issues/1006#issuecomment-400600105 (2., it's a bit saddening that it needs to be stated, but: 1. is not meant to be serious). It limits creating mentions to the help of auto-completion and ensures no technical data is expose while twitter handles an location hints stay in place.  Alas it brings a lot of work with it, bad for 14. 

A different, simpler approach would be to extend the display name resolver for users to throw an exception when the user was not found. Then the endpoint (we have the native DAV one an a custom talk one) to skip the mention, so it'll be formatted as plain text. That's rather quick and easy to accomplish, so the better option for 14. What I hate about this approach is that it still would expose technical data to end users (see implications above).

**Comment by MorrisJobke** [comment]
Moved to 15

**Comment by jancborchardt** [comment]
So what Github does is pretty simple … if the handle does not exist or does not exist _anymore_, it simply shows up as text like `@blizzzz`. Is this not something we could do?

**Comment by blizzz** [comment]
no, it'll show up like `@82308132-7140-435a-8c87-723b6b7ee069 ` or whatever the underlying backend uses as IDs.

**Comment by blizzz** [comment]
Also what Github does is renaming all deleted accounts to to @ghost which is maybe nicer than "@unknown user", but you'll just think it's just some person, then a very active person, and then you maybe look up the accounts page. Right now I am not sure they also change it in the comments text, but anywhere else in the UI.

**Comment by szaimen** [comment]
Is this still valid?
